mars/dataframe/core.py

```diff
@@ -487,7 +487,13 @@ def __repr__(self):
 
     @property
     def dtype(self):
-        return getattr(self, '_dtype', None) or self.op.dtype
+        # NB: don't use `getattr(self, '_dtype', None) or self.op.type`, since pd.Series has overloaded the
+        # operator `or`.
+        ty = getattr(self, '_dtype', None)
```

Comment:
I remember that I have encountered a related error when developing. However, no test cases can reproduce the error with the original implementation. It might be caused by an unrelated bug at other place and I have reverted this change.